sjsonnet/src/sjsonnet/Val.scala

```diff
@@ -40,16 +53,22 @@ object Val{
     * are all wrapped in [[Lazy]] and only truly evaluated on-demand
     */
   abstract class Lazy {
-    private[this] var cached: Val = null
+    protected[this] var cached: Val = null
     def compute(): Val
     final def force: Val = {
       if(cached == null) cached = compute()
       cached
     }
   }
+  final class Strict(v: Val) extends Lazy {
+    this.cached =v
```

Comment:
whitespace after the `=`

Reply:
I already tested all of these variants and they are all much slower.

The trick with setting `this.cached` lies in avoiding any megamorphic call when the `Lazy` gets evaluated. `force` is monomorphic, `compute` is megamorphic.

Reply:
Got it

Reply:
@szeiger although on further thought: couldn't we keep `.force` final, while still transitioning to the single type hierarchy to implement `.compute`? That would allow us to keep `force` monomorphic, keep `.compute` just as megamorphic as it is now, while still avoiding the overhead of boxing things in `Strict`

Reply:
It's also slower (although I don't fully understand why in this case)

Reply:
Got it. Hotspot moves in mysterious ways

Reply:
I tired it again and it actually works now. The last attempt (where I already used the trick of assigning the cached value in the constructor) was only a week ago. It's not a clear performance win but at least it's not slower and it allocates fewer objects.